### PR TITLE
Issue #3377397 by Kovalskiy266: Fix duplication of event enrollees and organizers

### PR DESCRIPTION
**Pull request by Kovalskiy266**
## Problem
If you add a translation to an event, it leads to duplicate enrollees and event organisers. This also applies to invitations and requests.

## Solution
Update the configurations of the views that are responsible for this, namely, enable the distinct option to prevent duplication.

## Issue tracker

- https://getopensocial.atlassian.net/browse/PROD-22011
- https://www.drupal.org/project/social/issues/3377397

## Theme issue tracker
N/A

## How to test
- [ ] As a sitemanager
- [ ] Create a new event and add translations to this event
- [ ] Enrol for this event
- [ ] Go to "Manage enrollments" page
- [ ] You will see that enrollees are duplicated
- [ ]  Also go to organisers page
- [ ] You will see that organisers are duplicated

### Before merge
- [ ] Code/peer review is completed
- [ ] All commit messages are [clear and clean](https://open-social.slite.com/app/docs/DnmermZDIx_0OQ). If applicable a rebase was performed
- [ ] All automated tests are green
- [ ] Functional/manual tests of the acceptance criteria are approved
- [ ] All acceptance criteria were met
- [ ] New features or changes to existing features are covered by tests, either unit (preferably) or behat
- [ ] Update path is tested. New hook_updates should respect update order, right naming convention and consider hook_post_update code
- [ ] Module can be safely uninstalled. Update/implement hook_uninstall and make sure that removed configuration or dependencies are removed/uninstalled
- [ ] This pull request has all required labels (team/type/priority)
- [ ] This pull request has a milestone
- [ ] This pull request has an assignee (if applicable)
- [ ] Any front end changes are tested on all major browsers
- [ ] New UI elements, or changes on UI elements are approved by the design team
- [ ] New features, or feature changes are approved by the product owner

### After merge
- [ ] Code is tested on all branches that it has been cherry-picked
- [ ] Update hook number might need adjustment, make sure they have the correct order
- [ ] The Drupal.org ticket(s) are updated according to this pull request status

## Screenshots
image
image

## Release notes
Fixes issue with duplication event enrollees and organisers, when event has several translations.

## Change Record
N/A

## Translations
N/A

### CODE REVIEW

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3471-lofqyd34sbwe90ua0lqebjnlhppc11oi.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/64c24511d8b2f31921bd548b

File: modules/social_features/social_event/social_event.install [hunk 1]
```diff
@@ -2295,3 +2295,32 @@ function social_event_update_11903(): void {
     \Drupal::logger('social_event')->error($e->getMessage());
   }
 }
+
+/**
+ * Update the views that are associated with enrolling in an event.
+ *
+ * To prevent duplication of users.
+ */
+function social_event_update_11904() {
```
Comment (ribel) [inline modules/social_features/social_event/social_event.install:2304]: @Kovalskiy266 Please change it also in the original config files, not only in the update hook.

**Review by ribel — CHANGES_REQUESTED** [review]

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3471-lofqyd34sbwe90ua0lqebjnlhppc11oi.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/64c24511d8b2f31921bd548b

File: modules/social_features/social_event/social_event.install (hunk 1 above)
Comment (Kovalskiy266) [reply modules/social_features/social_event/social_event.install]: Done

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3471-lofqyd34sbwe90ua0lqebjnlhppc11oi.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/64c24511d8b2f31921bd548b

**Comment by Kovalskiy266** [comment]
This also requires behat test.
So I'm in the process of doing that now.

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3471-lofqyd34sbwe90ua0lqebjnlhppc11oi.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/64c24511d8b2f31921bd548b

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3471-lofqyd34sbwe90ua0lqebjnlhppc11oi.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/64c24511d8b2f31921bd548b

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3471-lofqyd34sbwe90ua0lqebjnlhppc11oi.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/64c24511d8b2f31921bd548b

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3471-lofqyd34sbwe90ua0lqebjnlhppc11oi.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/64c24511d8b2f31921bd548b

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3471-lofqyd34sbwe90ua0lqebjnlhppc11oi.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/64c24511d8b2f31921bd548b

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3471-lofqyd34sbwe90ua0lqebjnlhppc11oi.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/64c24511d8b2f31921bd548b

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3471-lofqyd34sbwe90ua0lqebjnlhppc11oi.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/64c24511d8b2f31921bd548b

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3471-lofqyd34sbwe90ua0lqebjnlhppc11oi.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/64c24511d8b2f31921bd548b

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3471-lofqyd34sbwe90ua0lqebjnlhppc11oi.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/64c24511d8b2f31921bd548b

**Comment by Kovalskiy266** [comment]
@ribel Hi!
I've added scenario for checking if we have several translations of the event, the enrollees and organisers are not duplicated

**Review by ribel — COMMENTED** [review]
Hi @Kovalskiy266,
New scenario looks good to me, just add empty line in the end of test and squash last two commits into one, since they are related.

**Comment by Kovalskiy266** [comment]
> Hi @Kovalskiy266, New scenario looks good to me, just add empty line in the end of test and squash last two commits into one, since they are related.
Done

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3471-lofqyd34sbwe90ua0lqebjnlhppc11oi.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/64c24511d8b2f31921bd548b

**Review by ribel — APPROVED** [review]
All looks good to me now!

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3471-lofqyd34sbwe90ua0lqebjnlhppc11oi.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/64c24511d8b2f31921bd548b

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3471-lofqyd34sbwe90ua0lqebjnlhppc11oi.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/64c24511d8b2f31921bd548b

**Comment by BiaInacio** [comment]
Hi, I was not able to test because of the following error, it seams like an error on tugboat environment as it is trying to translate the images, if I understand the error message correctly, could any of you advise on how to fix it?
image
image

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3471-lofqyd34sbwe90ua0lqebjnlhppc11oi.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/64c24511d8b2f31921bd548b

**Comment by ribel** [comment]
Hi @BiaInacio
I see you tested it with the Brazilian Portuguese language, but I am not sure if we already support it in Open Social, so issues might be related to that. Let @Kovalskiy266 check if it was the case because previously all worked well.

**Comment by BiaInacio** [comment]
Hi @Kovalskiy266, Yes, I change the following configuration:
image
I enable both option for events, Show language selector on create and edit pages and Hide non translatable fields on translation forms, on /admin/config/regional/content-language. I believe this caused my issue as I was able to translate the content with them disable, but when I enable them I get the error I mention.
Is this an bug? If not, the change works fine! 😄

**Comment by Kovalskiy266** [comment]
@BiaInacio Yeah, activating "Hide non-translatable fields on translation forms" causes an error "Non-translatable fields can only be modified when updating the original language." But, it seems that the error with the generation of image is not related to this case.

**Comment by BiaInacio** [comment]
@Kovalskiy266 great, thank you for the explanation, all good then!

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3471-lofqyd34sbwe90ua0lqebjnlhppc11oi.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/64c24511d8b2f31921bd548b

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3471-lofqyd34sbwe90ua0lqebjnlhppc11oi.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/64c24511d8b2f31921bd548b

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3471-lofqyd34sbwe90ua0lqebjnlhppc11oi.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/64c24511d8b2f31921bd548b

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3471-lofqyd34sbwe90ua0lqebjnlhppc11oi.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/64c24511d8b2f31921bd548b

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3471-lofqyd34sbwe90ua0lqebjnlhppc11oi.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/64c24511d8b2f31921bd548b

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3471-lofqyd34sbwe90ua0lqebjnlhppc11oi.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/64c24511d8b2f31921bd548b

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3471-lofqyd34sbwe90ua0lqebjnlhppc11oi.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/64c24511d8b2f31921bd548b

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3471-lofqyd34sbwe90ua0lqebjnlhppc11oi.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/64c24511d8b2f31921bd548b

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3471-lofqyd34sbwe90ua0lqebjnlhppc11oi.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/64c24511d8b2f31921bd548b

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3471-lofqyd34sbwe90ua0lqebjnlhppc11oi.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/64c24511d8b2f31921bd548b

**Comment by ribel** [comment]
Hi @tbsiqueira and @nkoporec,
Since this was already tested - could you please check if I can merge it?

**Comment by Kingdutch** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3471-lofqyd34sbwe90ua0lqebjnlhppc11oi.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/64c24511d8b2f31921bd548b

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3471-lofqyd34sbwe90ua0lqebjnlhppc11oi.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/64c24511d8b2f31921bd548b

**Review by tbsiqueira — APPROVED** [review]

**Pull request merged**